meilisearch-release-changelog/v1.15.0.md

```diff
@@ -0,0 +1,63 @@
+# v1.15.0 release changelogs
+
+<!-- The following line should ONLY be put PRE-release changelogs -->
+> [!WARNING]
+> Since this is a release candidate (RC), we do NOT recommend using it in a production environment. Is something not working as expected? We welcome [bug reports](https://github.com/meilisearch/meilisearch/issues/new/choose) and [feedback about new features](https://github.com/meilisearch/product/discussions).
+
+Meilisearch v1.15.0 introduces improves the indexing performances and dumpless experience.
```

Comment:
Are they the things we want to promote?
-Dumpless upgrade is really niche to me (cf my comment under the section)
- performance improvements? about the new setting indexer?

We could promote: the new setting to Deactivate Typo Tolerance on Numbers and https://github.com/meilisearch/meilisearch/pull/5535